docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,163 @@
+
+## Proposed Solution
+
+#### 1. Version Picker
+
+Currently, the documentation is residing under the docs/ folder of Thanos repository. It is built by Hugo. It will have a proper drop-down menu just like prometheus [`drop-down menu`](https://prometheus.io/docs/introduction/overview/) which will enable proper versioning. This user facing tool will be built using HTML and CSS
+
+#### 2. Documentation Structure
+
+We want to propose a method called "Directory Sub Branching".
+Directory Sub branching means creating different sub branches in the `versioned-tmp` folder of the Thanos repository. Since the current architecture of the Thanos website is this:
+
+```|- website
+    |- archetypes
+    |- layout
+    |- data
+    |- static
+    |- resources
+    |- tmp
+        |- public
+        |- docs-pre-processed
+```
+We want to add aditional `versioned-tmp` folder within the website's `tmp` directory. For example:
+
+```|- website
+    |- archetypes
+    |- layout
+    |- data
+    |- static
+    |- resources
+    |- tmp
+        |- public
+        |- docs-pre-processed
+        |- versioned-tmp
+            |- master
+            |- version 0.13.0
+            |- version 0.12.2
+            |- other-folder-for-other-releases
+```
+_NOTE: `tmp` directory is not committed, just temporarily built. The current version of docs lives in the `master` folder_
+
+
+#### 3. Building a versioning plugin
+
+Creating a plugin that can automate these processes would save us alot of development time and stress. This approach promises to be very useful when it comes to versioning different release in the Thanos website.
+
+
+##### Workflow
+
+1. Developer makes all the necessary edit on `/docs` on master
+2. Developer proceeds by committing a new Release (i.e Release 0.x)
+3. CI run some `make web` or `Make web-serve` 
+4. Before anything, CI generate those docs and stores in a `versioned-tmp` folder
+5. rest of web command is done
+
+_NOTE: generated docs are not committed, just temporarily built._
+
+
+##### How to specify what part of docs you want to version without cloning every part of the docs? docs.yaml
+
+We hope to have a single flexible configuration file (`docs.yaml`) that will help the developer in specifying what part of docs he/she needs without cloning all the docs. We could have this as a single file on master. The config file will look like this:
+
+```versioned:
+   default:
+   - docs/components/*.md
+   - docs/storage.md
+   
+  overrides:
+     release-0.12:
+     - docs/storage2221241.md
+```
+##### Alternative path
+
+We could have this file on master, so the current `make web` will
+
+1. Parse on master `docs.yaml` and decide.
+1. Go to that particular release (`i.e release-0.10`)
+1. Is there docs.yaml? 
+    * Yes - parse and decide
+    * No - use docs.yaml from master
+
+The design of docs.yaml will look like this: 
+
+```versioned:
+   - docs/components/*.md
+   - docs/storage2221241.md
+```
+## FAQ
```

Comment:
I think this FAQ should be above the `##### How to specify what part of docs you want to version without cloning every part of the docs? docs.yaml` no? (:

Reply:
Okay...would apply changes ASAP

Reply:
Yes. I agree with you. FAQ should be above not below...